pkg/cmd/server/api/types.go

```diff
@@ -646,3 +646,75 @@ type PodManifestConfig struct {
 	// The interval needs to be a positive value
 	FileCheckIntervalSeconds int64
 }
+
+type LDAPSyncConfig struct {
+	// URL is a partial RFC 2255 URL which specifices the scheme, host and port of the LDAP server
+	// The syntax of the URL is: `scheme://host:port/`, where `scheme` is either ldap or ldaps
+	URL string
+
+	// Insecure, if true, indicates the connection should not use TLS. Cannot be set to true with a
+	// URL scheme of "ldaps://" If false, "ldaps://" URLs connect using TLS, and "ldap://" URLs are
+	// upgraded to a TLS connection using StartTLS as specified in https://tools.ietf.org/html/rfc2830
+	Insecure bool
+
+	// CA is the optional trusted certificate authority bundle to use when making requests to the server
+	// If empty, the default system roots are used
+	CA string
+
+	// Attributes maps LDAP attributes to identities
+	Attributes LDAPAttributes
+
+	// GroupNameAttributes (optionally) maps LDAP group entries to OpenShift Groups
+	GroupNameAttributes []string
+
+	// UserDefinedGroupNameMapping determines if the mapping of LDAP group UIDs to OpenShift Group
+	// names is defined by an input mapping or algorithmically
+	UserDefinedGroupNameMapping bool
+
+	// LDAPGroupUIDToOpenShiftNameMapping is an optional direct mapping of LDAP group UIDs to OpenShift
+	// Group names
+	LDAPGroupUIDToOpenShiftNameMapping map[string]string
+
+	// GroupQueryTemplate holds the (optional) template for an LDAP query that returns group entries
+	GroupQueryTemplate LDAPQueryTemplate
+
+	// UserQueryTemplate holds the template for an LDAP query that returns user entries
+	UserQueryTemplate LDAPQueryTemplate
+}
+
+type LDAPQueryTemplate struct {
```

Comment:
Depending on the schema, the type of template (are we grabbing something specific? do we need to insert something from the attribute field of another LDAP entry) will change. I couldn't figure out how to unite the schema-specific query templates for groups and users without this ugly superset impl. Suggestions welcome.